src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -133,8 +135,16 @@ public String tagImageURLs(String message)
       
       if(isUrlValid)
       {
-        String replacement = "<img src=\"$1\" />";
-        urls.set(i, urls.get(i).replaceAll(regex, replacement));
+        String imageReplacement = "<img src=\"$1\" />";
```

Comment:
Nit: since these are string constants that don't change between calls, consider making them private static final class variables:
https://www.geeksforgeeks.org/final-static-variable-java/